### ISSUE TITLE
Fix `unit-no-unknown` false negatives with option `ignoreFunctions`

### ISSUE THREAD

**Issue by romainmenke** [stylelint/stylelint#6888]
### What steps are needed to reproduce the bug?

```css
a {
	margin: foo(calc(10pixels));
}
```

### What Stylelint configuration is needed to reproduce the bug?

```json
{
  "rules": {
    "unit-no-unknown": [true, { "ignoreFunctions": ["foo"] }]
  }
}
```

### How did you run Stylelint?

https://stylelint.io/demo/#N4Igxg9gJgpiBcICGACYAdAdugLgWyQCcBzAS03hQDMIIAKMJAGzDoEYAGAB1IA8YmAZwCUwgNxYAviAA0IKqSYwAckjxxEMXmq5KAdGEGDZ4CJgXEEIDJhQp0IQgFclgh5Rt27Dp5lI4AWkwIAN8Aa2CAd2wEFABtHGcYGTR7EFJiYMIYADFfMBxSMzdYuIcaCAcAXRRJKqw7SSkTSHMMnIhCAhwrACtBMxNYLmNETzTBHABPJSZyHHc0piQcGEmHGQaJ6dn5gNaLAMmkTCgiKEWHZdX1kGbJIA

### Which version of Stylelint or dependencies are you using?

15.6.3

### What did you expect to happen?

I expected an error.
The same error as with : https://stylelint.io/demo/#N4Igxg9gJgpiBcICGACYAdAdugLgWyQCcBzAS03hTCQBswAKARgAYAHUgDxhoGcBKANxYAviAA0IAGakaMAHJI8cRDA6LWsgHRgePceAiZpxBCAyYUKdCEIBXWT2uVzly9duZSOALSYI3jwBrPwB3bAQUAG0cOxgxNCsQUmI-QhgAMQ8wHFJDRwjI60kICGsAXRRhMqxLYRF9SCNk9IhCAhxTACseQ31YVj1EF0SeHABPWRpyHCdEmiQcGFHrMRqR8cnp70bjb1GkTCgiKFnrecXlkHrhIA

```css
a {
	margin: calc(10pixels);
}
```

### What actually happened?

No error.

### Proposal to fix the bug

Correctly track the state for ignoring units within functions.

**Comment by romainmenke** [comment]
I was mistaken.

The current behavior matches older versions : https://stylelint.io/demo/#N4Igxg9gJgpiBcICGACYAdAdugLgWyQCcBzAS03hQDMIIAKMJAGzDoEYAGAB1IA8YmAZwCUwgNxYAviAA0IKqSYwAckjxxEMXmq5KAdGEGDZ4CJgXEEIDJhQp0IQgFclgh5Rt27Dp5lI4AWkwIAN8Aa2CAd2wEFABtHGcYGTR7EFJiYMIYADFfMBxSMzdYuIcaCAcAXRRJKqw7SSkTSHMMnIhCAhwrACtBMxNYLmNETzTBHABPJSZyHHc0tgBWPVWOBxkGienZ+YDWiwDJpEwoIihFhyYkHBhJh2bJIA
